dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/csv/CsvEventDataValue.java

```diff
@@ -0,0 +1,497 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.tracker.csv;
+
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+/**
+ * @author Enrico Colasante
+ */
+@JsonPropertyOrder( {
+    "event",
+    "status",
+    "program",
+    "programStage",
+    "enrollment",
+    "orgUnit",
+    "orgUnitName",
+    "occurredAt",
+    "scheduledAt",
+    "latitude",
+    "longitude",
+    "followup",
+    "deleted",
+    "createdAt",
+    "createdAtClient",
+    "updatedAt",
+    "updatedAtClient",
+    "completedBy",
+    "completedAt",
+    "updatedBy",
+    "attributeOptionCombo",
+    "attributeCategoryOptions",
+    "assignedUser",
+    "dataElement",
+    "value",
+    "storedBy",
+    "providedElsewhere",
+    "completedAt",
+    "completedBy",
+    "geometry"
+} )
+public class CsvEventDataValue
+{
+    private String event;
+
+    private String status;
+
+    private String program;
+
+    private String programStage;
+
+    private String orgUnit;
+
+    private String orgUnitName;
+
+    private String enrollment;
+
+    private String occurredAt;
+
+    private String scheduledAt;
+
+    private boolean followup;
+
+    private boolean deleted;
+
+    private String createdAt;
+
+    private String createdAtClient;
+
+    private String updatedAt;
+
+    private String updatedAtClient;
+
+    private String completedBy;
+
+    private String completedAt;
+
+    private String updatedBy;
+
+    private Double latitude;
+
+    private Double longitude;
+
+    private String attributeOptionCombo;
+
+    private String attributeCategoryOptions;
+
+    private String assignedUser;
+
+    private String dataElement;
+
+    private String value;
+
+    private String storedBy;
+
+    private Boolean providedElsewhere;
+
+    private String createdAtDataValue;
+
+    private String updatedAtDataValue;
+
+    private String geometry;
+
+    public CsvEventDataValue()
+    {
+    }
+
+    public CsvEventDataValue( CsvEventDataValue dataValue )
+    {
+        Assert.notNull( dataValue, "A non-null CsvOutputEventDataValue must be given as a parameter." );
```

Comment:
I often see us using https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-
Do we care if it's an IllegalArugmentException vs NullpointerException? If not I would tend to use the Java one instead of Springs util

Reply:
I think it is just to give a cleaner message